app/models/user.rb

```diff
@@ -0,0 +1,18 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  before_save :downcase_email
+
+  validates :name, presence: true,
+length: {maximum: Settings.model.validate_name}
+  validates :email, presence: true,
+  length: {maximum: Settings.model.validate_email}
```

Comment:
Chỗ này rubocop k bắt lỗi à ta
```suggestion
  validates :name, presence: true, length: {maximum: Settings.model.validate_name}
  validates :email, presence: true, length: {maximum: Settings.model.validate_email}
```
hoặc
```suggestion
  validates :name, presence: true,
             length: {maximum: Settings.model.validate_name}
  validates :email, presence: true,
             length: {maximum: Settings.model.validate_email}
```